src/core/device.c

```diff
@@ -639,13 +685,9 @@ static void device_found_changed(Device *d, DeviceFound previous, DeviceFound no
 }
 
 static void device_update_found_one(Device *d, DeviceFound found, DeviceFound mask) {
-        Manager *m;
-
         assert(d);
 
-        m = UNIT(d)->manager;
-
-        if (MANAGER_IS_RUNNING(m) && (m->honor_device_enumeration || MANAGER_IS_USER(m))) {
+        if (MANAGER_IS_RUNNING(UNIT(d)->manager)) {
```

Comment:
This drops the condition `m->honor_device_enumeration`. Thus if the udev db is not initialized, `d->enumerated_found` will be `DEVICE_NOT_FOUND`. Thus `manager_startup() → manager_ready() → unit_catchup() → device_catchup() → device_update_found_one() will see `MANAGER_OK`, and will reset `d->found` to `d->enumerated_found == DEVICE_FOUND_NONE`, with possibly fatal consequences. What am I missing here?

Reply:
IOW, I think the test for `m->honor_device_enumeration` must be kept here.

Reply:
`m->honor_device_enumeration` matters only if switching root. Following a similar argument as https://github.com/systemd/systemd/pull/23218#issuecomment-1114901938, the device would have been reset to `DEVICE_TENTATIVE` beforehand. If the device had been mounted before switching root, `device_enumerate()` should have seen it and `enumerated_found` should be `DEVICE_FOUND_MOUNTED`.
So I *suppose* this change does no harm. It's the kind of thing that scares me a bit though, because of past experience with the subtleties of the device handling code.

As `m->honor_device_enumeration` now doesn't affect any more whether device enumeration is honored, this change is definitely an argument for changing the field name from `m->honor_device_enumeration` to `m->honor_serialized_device_state` or perhaps simply `m->switching_root`.

Reply:
My concern here was that `device_coldplug()` now calls `device_set_state()` even if `honor_device_enumeration` is false. But as I said previously, I believe it's ok, and my testing yielded favourable results. This conversation can be considered resolved.

>  As I commented in the code, on switching root, `device_enumerate()` lists almost no devices, as udev database is cleared. 

FTR: true,  but `manager_enumerate()` also calls `mount_enumerate()`, which may set `DEVICE_FOUND_MOUNT`.